tests/Application/Commands/DeleteStalePaymentDetailsTest.php

```diff
@@ -0,0 +1,178 @@
+        $testPaymentMethodFingerprint = 'Xt5EWLLDS7FJjR1c';
+
+        $stripeChargesProphecy = $this->prophesize(ChargeService::class);
+        $stripeChargesProphecy->search([
+            'query' => sprintf(
+                'customer:"%s" and payment_method_details.card.fingerprint:"%s" and status:"succeeded"',
+                $testCustomerId,
+                $testPaymentMethodFingerprint,
+            ),
+            'limit' => 100,
+        ])
+            ->shouldBeCalledOnce()
+            ->willReturn($this->buildCollectionFromSingleObjectFixture('ch_succeeded'));
+
+        $stripePaymentMethodsProphecy = $this->prophesize(PaymentMethodService::class);
+        $stripePaymentMethodsProphecy->all([
+            'customer' => $testCustomerId,
+            'type' => 'card',
+            'limit' => 100,
+        ])
+            ->shouldBeCalledOnce()
+            ->willReturn($this->buildCollectionFromSingleObjectFixture(
+                $this->getStripeHookMock('ApiResponse/pm'),
+            ));
+
+        // *No* PM should be detached i.e. soft deleted, with any ID.
+        $stripePaymentMethodsProphecy->detach(Argument::type('string'))
+            ->shouldNotBeCalled();
+
+        $oneDayBeforeTest = $initDate
+            ->sub(new \DateInterval('P1D'))
+            ->getTimestamp();
+
+        $stripeCustomersProphecy = $this->prophesize(CustomerService::class);
+        $stripeCustomersProphecy->search([
+            'query' => "created<$oneDayBeforeTest and metadata['hasPasswordSince']:null",
+            'limit' => 100,
+        ])
+            ->shouldBeCalledOnce()
+            ->willReturn($this->buildCollectionFromSingleObjectFixture(
+                $this->getStripeHookMock('ApiResponse/customer'),
+            ));
+
+        $stripeClientProphecy = $this->prophesize(StripeClient::class);
+
+        $stripeClientProphecy->charges = $stripeChargesProphecy->reveal();
+        $stripeClientProphecy->customers = $stripeCustomersProphecy->reveal();
+        $stripeClientProphecy->paymentMethods = $stripePaymentMethodsProphecy->reveal();
+
+        $commandTester = new CommandTester($this->getCommand(
+            $stripeClientProphecy,
+            $initDate,
+        ));
+        $commandTester->execute([]);
+
+        $expectedOutputLines = [
+            'matchbot:delete-stale-payment-details starting!',
+            'Deleted 0 payment methods from Stripe, having checked 1 customers',
+            'matchbot:delete-stale-payment-details complete!',
+        ];
+        $this->assertEquals(implode("\n", $expectedOutputLines) . "\n", $commandTester->getDisplay());
+
+        $this->assertEquals(0, $commandTester->getStatusCode());
+    }
+
+    private function getCommand(
+        ObjectProphecy $stripeClientProphecy,
+        \DateTimeImmutable $initDate,
+    ): DeleteStalePaymentDetails {
+        $stripeClient = $stripeClientProphecy->reveal();
+        \assert($stripeClient instanceof StripeClient);
```

Comment:
You can replace the assert with a docblock if you like:

```suggestion
    /**
     * @param ObjectProphecy<StripeClient> $stripeClientProphecy
     */
    private function getCommand(
        ObjectProphecy $stripeClientProphecy,
        \DateTimeImmutable $initDate,
    ): DeleteStalePaymentDetails {
        $stripeClient = $stripeClientProphecy->reveal();
```